commercetools-sdk-base/src/test/java/io/sphere/sdk/queries/PagedQueryResultTest.java

```diff
@@ -44,6 +46,32 @@ public void middleOfManyResult() throws Exception {
     }
 
     @Test
+    public void getResultPage() throws Exception {
+        final PagedQueryResult<Integer> queryResult = PagedQueryResult.of((long) 80, TOTAL, listOfSize(PAGE_SIZE));
+        assertThat(queryResult.getPage()).isEqualTo(3);
```

Comment:
The data is a bit unrealistic so it's difficult to know the page it should be. If you had offset 80 and total products of 100, then you should not be receiving 25 products, as 80 is not divisible by 25. So it's difficult to determine if the test is good.
So let's say that you are getting 25 products back because you are neither at the 1st or the last page. Then we could say we requested an offset of 75 for example, in which case the page should be 3 (starting from 0).
In general I would recommend to find edge cases, not based on your implementation, but on the problem you are solving. With pages that could be:
a) request didn't raise any results at all (offset = 0, total = 0, count = 0, page = 0)
b) request has more than one page and ends with a count < page size:
   1. request first page (offset = 0, total = 106, count = 25, page = 0)
   2. request middle page (offset = 75, total = 106, count = 25, page = 3)
   3. request last page (offset = 100, total = 106, count = 6, page = 4)
   4. request past the last page (offset = 125, total = 106, count = 0, page = ?? as we don't know the requested page size we cannot know the page, but returning 0 can be misleading...)

Pagination might look like easy to implement, but it's very tricky. We should better have strong tests behind it :)
And I guess we should assume the backend doesn't return inconsistent results, but it's worth asking them if that can happen. With facets that could happen, for example.